yul_to_ir/src/instruction.rs

```diff
@@ -0,0 +1,1183 @@
+                    .get(0)
+                    .ok_or(ASTLoweringError {
+                        message: "the number of args of instr \"mstore8\" is less than 2 "
+                            .to_string(),
+                    })
+                    .cloned()?;
+                let value = args
+                    .get(1)
+                    .ok_or(ASTLoweringError {
+                        message: "the number of args of instr \"mstore8\" is less than 2 "
+                            .to_string(),
+                    })
+                    .cloned()?;
+                self.build_data_store_byte(data, offset, value)
+            }
+            YulInstructionName::SLoad => {
+                let storage_path = self.ir_context.builder.instr_get_storage_path(args);
+                Ok(self
+                    .ir_context
+                    .builder
+                    .instr_storage_load(storage_path.into(), WORD_TY)
+                    .into())
+            }
+            YulInstructionName::SStore => {
+                let pos = args
+                    .get(0)
+                    .ok_or(ASTLoweringError {
+                        message: "the number of args of instr \"sstore\" is less than 2 "
+                            .to_string(),
+                    })
+                    .cloned()?;
+                let value = args
+                    .get(1)
+                    .ok_or(ASTLoweringError {
+                        message: "the number of args of instr \"sstore\" is less than 2 "
+                            .to_string(),
+                    })
+                    .cloned()?;
+                let storage_path = self.ir_context.builder.instr_get_storage_path(vec![pos]);
+                Ok(self
+                    .ir_context
+                    .builder
+                    .instr_storage_store(storage_path.into(), value)
+                    .into())
+            }
+            YulInstructionName::MSize => Ok(self
+                .ir_context
+                .builder
+                .instr_div(
+                    self.ir_context
+                        .builder
+                        .instr_call(
+                            get_intrinsic_func_by_key(IntrinsicFuncName::IR_VECTOR_LEN)
+                                .unwrap()
+                                .into(),
+                            vec![self
+                                .ir_context
+                                .builder
+                                .build_identifier(&self.data_id.borrow())],
+                            WORD_TY,
+                        )
+                        .into(),
+                    self.hex_literal(&WORD_SIZE.to_string()),
+                )
+                .into()),
+            YulInstructionName::Gas => todo!(),
+            YulInstructionName::Address => todo!(),
+            YulInstructionName::Balance => todo!(),
+            YulInstructionName::SelfBalance => todo!(),
```

Comment:
could you pls reorder all todo!() macros to gather them together in the end of the switch case